src/java.base/share/classes/java/util/jar/JarFile.java

```diff
@@ -837,26 +837,34 @@ private void initializeVerifier() {
     public synchronized InputStream getInputStream(ZipEntry ze)
         throws IOException
     {
-        maybeInstantiateVerifier();
-        if (jv == null) {
-            return super.getInputStream(ze);
-        }
-        if (!jvInitialized) {
-            initializeVerifier();
-            jvInitialized = true;
-            // could be set to null after a call to
-            // initializeVerifier if we have nothing to
-            // verify
-            if (jv == null)
+        Objects.requireNonNull(ze, "ze");
+        try {
+            maybeInstantiateVerifier();
+            if (jv == null) {
                 return super.getInputStream(ze);
-        }
+            }
+            if (!jvInitialized) {
+                initializeVerifier();
+                jvInitialized = true;
+                // could be set to null after a call to
+                // initializeVerifier if we have nothing to
+                // verify
+                if (jv == null)
+                    return super.getInputStream(ze);
+            }
 
-        // wrap a verifier stream around the real stream
-        return new JarVerifier.VerifierStream(
-            getManifestFromReference(),
-            verifiableEntry(ze),
-            super.getInputStream(ze),
-            jv);
+            // wrap a verifier stream around the real stream
+            return new JarVerifier.VerifierStream(
+                    getManifestFromReference(),
+                    verifiableEntry(ze),
+                    super.getInputStream(ze),
+                    jv);
+        } catch (IOException | IllegalStateException | SecurityException e) {
+            throw e;
+        } catch (Exception e2) {
+            // Any other Exception should be a ZipException
+            throw (ZipException) new ZipException("Zip file format error").initCause(e2);
```

Comment:
If there is ZIP format error then I would expect ZipException or the more general IOException is already thrown. So I think this is catching other cases, maybe broken manifests or signed JAR files, in which case a JarException may be better.

Reply:
JarFile::getInputStream. mentions ZipException but not JarException which is why I chose this.  If we change this to JarException, I would need to update the javadoc and create a CSR.

Please let me know your preference

Reply:
If you are pretty sure the only other case are as above, I wonder if a simpler fix would be to change `verifiableEntry()` to check for these null cases and throw a `ZipException` which will get directly propagated by `getInputStream()`?

Reply:
I can certainly throw a ZipException from `verifiableEntry`. I am a bit reluctant to not catch any other `Exception` and then throw a `ZipException` from `getInputStream()` as it is certainly possible of encountering some other issue due to some stray value in the CEN.

So I will update `verifiableEntry` to validate `ZipEntry` and` ZipEntry::getName()` potential issues

Reply:
Per an offline-discussion with Sean, I narrowed the Exception checking to JarFile::verifiableEntry.